pdns/ixfr.cc

```diff
@@ -168,8 +168,12 @@ vector<pair<vector<DNSRecord>, vector<DNSRecord> > > getIXFRDeltas(const ComboAd
   std::shared_ptr<SOARecordContent> masterSOA = nullptr;
   vector<DNSRecord> records;
   size_t receivedBytes = 0;
+  int8_t ixfrInProgress = -2;
 
   for(;;) {
+    if (!ixfrInProgress)
```

Comment:
This test would fail if we somehow received a transaction containing the last SOA record _thrice_. I agree this shouldn't happen, but how about testing `if (ixfrInProgress >= 0)` instead?

Reply:
Yeah agree with that. It's more protective.